src/Raven.Server/Documents/DocumentPutAction.cs

```diff
@@ -162,9 +162,12 @@ public virtual void ValidateId(Slice lowerId)
                     // anything else - must match exactly
 
                     oldChangeVector = TableValueToChangeVector(context, (int)DocumentsTable.ChangeVector, ref oldValue);
+                    
                     if (expectedChangeVector != null)
                     {
-                        if (string.Compare(expectedChangeVector, oldChangeVector, StringComparison.Ordinal) != 0)
+                        var expected = context.GetChangeVector(expectedChangeVector);
+
+                        if (string.Compare(expected.Version, oldChangeVector.Version, StringComparison.Ordinal) != 0)
```

Comment:
use ChangeVector.CompareVersion ?

It is always better to use a dedicated method, because it is easier then to track down all the places that require that functionality and fix it in on shot if needed

Reply:
I wasn't aware that we have `ChangeVector.CompareVersion` method, thanks
fixed